src/compat/posix/time/strptime.c

```diff
@@ -0,0 +1,327 @@
+/**
+ * @file strptime.c
+ * @brief writing linux function strptime for embox os
+ *
+ * author Tamara Deryugina <tamara.deryugina@gmail.com>
+ * @date 02.03.2020
+ */
+
+#define _XOPEN_SOURCE
+#include<stdio.h>
+#include<stdlib.h>
+#include<string.h>
+#include<time.h>
+#include<ctype.h>
+#include<math.h>
+
+int str_legth(const char *s) {
+	int count = 0;
+	while(s[count] != '\0') {
+		count++;
+	}
+	return count;
+ }
+
+int num_length(char *new_s) {
+	int count = 0;
+	while ((new_s[count] > 47) && (new_s[count] < 58)) {
+		count++;
+	}
+	return count;
+}
+
+int word_length(const char *s) {
+	int count = 0;
+	while ((s[count] > 96) && (s[count] < 123)) {
+		count++;
+	}
+	return count;
+}
+
+int degree(int dec, int num) {
+	int count = 0;
+	int res = 1;
+	if (num == 0) {
+		return 1;
+	}
+	else if (num == 1) {
+		return dec;
+	}
+	while (count != num) {
+		res *= dec;
+		count++;
+	}
+	return res;
+}
+
+int transform_to_int(char *new_s, int num) {
+	int count = 0;
+	int a, b = 0;
+	int c;
+	while ((new_s[count] > 47) && (new_s[count] < 58)) {
+		c = degree(10, num-1);
+		a = (new_s[count] - '0')*c;
+		b+=a;
+		count++;
+		num--;
+	}
+	return b;
+}
+
+char *cmp_before_procent(char *new_s, char *new_form) {
+	int count = 0;
+	while (new_form[count] != '\0') {
+		if (new_form[count] == '%') {
+			return &new_form[count];
+		}
+		if ((new_form[count] > 64) && (new_form[count] < 91)) {
+			new_form[count] = tolower(new_form[count]);
+		}
+		if (strncmp(&new_form[count], &new_s[count], 1) != 0) {
+			return NULL;
+		}
+		count++;
+	}
+	return &new_form[count];
+}
+
+char *def_format(char *new_form, char *new_s, struct tm *tm) {
+
+	char arr_day[7][10] = {"sunday", "monday", "tuesday", "wednesday", "thursday", "friday", "saturday"};
+	char arr_month[6][10] = {"january", "february", "march", "april", "may", "june"};
```

Comment:
Why do you split months into two arrays?

Reply:
Yes because it's too long for one line

Reply:
You can split it, just make newline after another comma sign.
```
int a = {0, 1,
     2, 3};
```

Reply:
I fixed this line.
Thanks!